dbt-teradata 1.8.0 changes (#5677)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/teradata-configs.md

        +use_fastload: true
    ```

## Snapshots

> Snapshots uses the HASHROW function of the Teradata database to generate a unique hash value for the 'dbt_scd_id' column. If you want to use your own hash UDF, there is a configuration option in the snapshot model called 'snapshot_hash_udf', which defaults to HASHROW. You can provide a value like <database_name.hash_udf_name>. If only hash_udf_name is provided, it uses the same schema as the model runs.

Reviewer: ```suggestion

Snapshots use the [HASHROW function](https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Functions-Expressions-and-Predicates/Hash-Related-Functions/HASHROW/HASHROW-Function-Syntax) of the Teradata database to generate a unique hash value for the `dbt_scd_id` column. 

To use your own hash UDF, there is a configuration option in the snapshot model called `snapshot_hash_udf`, which defaults to HASHROW. You can provide a value like `<database_name.hash_udf_name>`. If you only provide `hash_udf_name`, it uses the same schema as the model runs.
```

        +use_fastload: true
    ```

## Snapshots

Snapshots use the [HASHROW function](https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Functions-Expressions-and-Predicates/Hash-Related-Functions/HASHROW/HASHROW-Function-Syntax) of the Teradata database to generate a unique hash value for the `dbt_scd_id` column. 

To use your own hash UDF, there is a configuration option in the snapshot model called `snapshot_hash_udf`, which defaults to HASHROW. You can provide a value like `<database_name.hash_udf_name>`. If you only provide `hash_udf_name`, it uses the same schema as the model runs.

> for e.g. :

Reviewer: ```suggestion
For example, in the `snapshots/snapshot_example.sql` file:

```

        +use_fastload: true
    ```

## Snapshots

Snapshots use the [HASHROW function](https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Functions-Expressions-and-Predicates/Hash-Related-Functions/HASHROW/HASHROW-Function-Syntax) of the Teradata database to generate a unique hash value for the `dbt_scd_id` column. 

To use your own hash UDF, there is a configuration option in the snapshot model called `snapshot_hash_udf`, which defaults to HASHROW. You can provide a value like `<database_name.hash_udf_name>`. If you only provide `hash_udf_name`, it uses the same schema as the model runs.

For example, in the `snapshots/snapshot_example.sql` file:

>   snapshots/snapshot_example.sql

Reviewer: ```suggestion
```

  ```
> :information_source: `copy_grants` is not supported in Teradata.

Refer to [grants](/reference/resource-configs/grants) for more information on Grants.

## Query Band

> Query Band in dbt-teradata can be set on three levels:

Reviewer: adjusted this section as the formatting was off

<img width="812" alt="Screenshot 2024-07-02 at 12 22 43" src="https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/7bd654df-0b18-489f-b295-e89c1d0035e8">

   ```

2. Project level: In the `dbt_project.yml` file, the user can provide `query_band` using the following example:

   ```yaml
     models:
     Project_name:
        +query_band: "app=dbt;model={model};"
   ```
4. Model level: It can be set on the model SQL file or model level configuration on YAML files:

   ```sql
   {{ config( query_band='sql={model};' ) }}
   ```

Users can set `query_band` at any level or on all levels. With profiles-level `query_band`, dbt-teradata will set the `query_band` for the first time for the session, and subsequently for model and project level query band will be updated with respective configuration.

If a user sets some key-value pair with value as `'{model}'`, internally this `'{model}'` will be replaced with model name, which can be useful for telemetry tracking of sql/ dbql logging. 

  ```yaml
  models:
  Project_name:
    +query_band: "app=dbt;model={model};"
  ````

- For example, if the model the user is running is `stg_orders`, `{model}` will be replaced with `stg_orders` in runtime.
- If no `query_band` is set by the user, the default query_band used will be: ```org=teradata-internal-telem;appname=dbt;```

> ## valid_history incremental materialization strategy (early access) {valid_history-incremental-materialization-strategy}

Reviewer: ```suggestion
## valid_history incremental materialization strategy (early access) {#valid-history-incremental-materialization-strategy}
```

> ## valid_history incremental materialization strategy (early access) {#valid-history-incremental-materialization-strategy}

Reviewer: suggesting removing the 'early access' from the header

```suggestion
## valid_history incremental materialization strategy
_This is available in early access_
```

> More on Grants can be found at https://docs.getdbt.com/reference/resource-configs/grants

Reviewer: ```suggestion
Refer to [grants](/reference/resource-configs/grants) for more information on Grants.
```
